src/Umbraco.Core/Cache/CacheRefresherNotificationBase.cs

```diff
@@ -0,0 +1,83 @@
+﻿using Umbraco.Cms.Core.Events;
+using Umbraco.Cms.Core.Sync;
+
+namespace Umbraco.Cms.Core.Cache
+{
+    /// <summary>
+    /// Event args for cache refresher updates
+    /// </summary>
+    public abstract class CacheRefresherNotificationBase : INotification
+    {
+        public CacheRefresherNotificationBase Init(object msgObject, MessageType type)
```

Comment:
Can we do better than this `Init` pattern? I realize it's an easy way to have the generic usage with the `new()` constraint but could be achieved with a factory instead. I've pushed a change for this along with splitting the classes into separate files. It's now just a `ICacheRefresherNotificationFactory` that creates the notifications and seems to work well.  Let me know if you disagree. See rev: f03fc25b4c49718218c1d0da64739b2aa1714d10